### PR TITLE
fix issue 5714.

### PR DESCRIPTION
**Pull request by yuyang-ok**
👏 hey.

This fix https://github.com/bytecodealliance/wasmtime/issues/5714
I make a mistaken about the tmp register.
I think it is an int register, it is actual a float register.

### CODE REVIEW

File: cranelift/filetests/filetests/runtests/issue5714.clif [hunk 1]
```diff
@@ -0,0 +1,23 @@
+test interpret
+test run
+target riscv64gc
```
Comment (afonso360) [inline cranelift/filetests/filetests/runtests/issue5714.clif:3]: Would you be able to add targets for the other architectures? Even though this issue didn't impact them, It's still a useful regression test!

**Review by afonso360 — APPROVED** [review]
LGTM! I'm not 100% sure about the regalloc changes, if someone else could double check that, it would be nice.

My understanding of this (based on Chris's great [blog post](https://cfallin.org/blog/2022/06/09/cranelift-regalloc2/)) is that since we write to `rd` in the middle of the pseudo-instruction we need mark it as early def right?

File: cranelift/filetests/filetests/runtests/issue5714.clif (hunk 1 above)
Comment (yuyang-ok) [reply cranelift/filetests/filetests/runtests/issue5714.clif:3]: sure.

**Comment by yuyang-ok** [comment]
> LGTM! I'm not 100% sure about the regalloc changes, if someone else could double check that, it would be nice.
I think regalloc changes work same  In current situation, I think `tmp` register used multi time usually, better `reg_early_def`.

**Review by cfallin — APPROVED** [review]
LGTM on regalloc changes -- it looks like `rd` starts to be used before the last use of `tmp` at least, so its live-range overlaps, so it needs to be an early-def: so this is correct.

**Comment by alexcrichton** [comment]
I think the failure in [this test run](https://github.com/bytecodealliance/wasmtime/actions/runs/4256631285/jobs/7405760281) was legitimate, namely the new `*.clif` test added in this PR is panicking on the x64 backend. I think it's ok to remove the `target x86_64` line and file an issue to let this PR land.

**Comment by jameysharp** [comment]
I can confirm that the new CLIF test fails on x86_64 while emitting machine code for the `CvtFloatToSintSeq` pseudoinstruction. I think none of the float-to-int variants (saturating or trapping, signed or unsigned) are implemented in the x64 backend for types other than I32 and I64, but I'm not absolutely sure I read the ISLE rules correctly.

**Comment by yuyang-ok** [comment]
@alexcrichton I have remove test for `x86_64`.

**Comment by jameysharp** [comment]
@yuyang-ok, thank you!

**Pull request merged**